File: Cyventures/TombsEditor/ConfirmQuitState.cs
﻿using Common;
using EditorCommon;
using MonoGameCommon;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TombsEditor
{
    public class ConfirmQuitState : CyMenuState<EditorState>
    {
        private static readonly List<string> _items = new List<string>() { "No", "Yes" };

        public ConfirmQuitState(StateManager<EditorState, Command> manager, ColorBuffer<CyColor> screen, CyFont font)
            : base(manager, screen, font, "Are you sure?", _items)
        {
        }
        public override void OnItemSelected(int itemIndex)
        {
            switch(itemIndex)
            {
                case 0:
                    Manager.Set(EditorState.MainMenu);
                    break;
                case 1:
                    Manager.Set(EditorState.Quit);
                    break;
            }
        }
    }
}

File: web/scripts/ui/boilerplate/confirmQuitState.js
import {Display} from "../../common/Display.js";
import {MainMenuState} from "./mainMenuState.js";
import SplashState from "./splashState.js";

export class ConfirmQuitState {
    static run(){
        Display.clear();
        Display.addSimpleChild("p", "Are you sure you want to quit?");
        Display.addButton("Yes", ConfirmQuitState.confirm);
        Display.addButton("No", ConfirmQuitState.cancel)
    }
    static confirm(){
        SplashState.run();
    }
    static cancel(){
        MainMenuState.run();
    }
}
